tests/aws/services/apigateway/test_apigateway_s3.py

```diff
@@ -0,0 +1,224 @@
+    )
+
+    aws_client.apigateway.put_method_response(
+        restApiId=api_id, resourceId=delete_resource_id, httpMethod="GET", statusCode="200"
+    )
+    aws_client.apigateway.put_method_response(
+        restApiId=api_id, resourceId=put_resource_id, httpMethod="GET", statusCode="200"
+    )
+    aws_client.apigateway.put_method_response(
+        restApiId=api_id, resourceId=get_resource_id, httpMethod="GET", statusCode="200"
+    )
+
+    aws_client.apigateway.put_integration(
+        restApiId=api_id,
+        resourceId=get_resource_id,
+        httpMethod="GET",
+        integrationHttpMethod="GET",
+        type="AWS",
+        uri=f"arn:aws:apigateway:{region_name}:s3:path/{bucket}/{object_name}",
+        credentials=role_arn,
+    )
+    aws_client.apigateway.put_integration(
+        restApiId=api_id,
+        resourceId=put_resource_id,
+        httpMethod="GET",
+        integrationHttpMethod="PUT",
+        type="AWS",
+        uri=f"arn:aws:apigateway:{region_name}:s3:path/{bucket}/{object_name}",
+        requestParameters={
+            "integration.request.header.Content-Type": "'application/json'",
+        },
+        requestTemplates={"application/json": '{"message": "great success!"}'},
+        credentials=role_arn,
+    )
+    aws_client.apigateway.put_integration(
+        restApiId=api_id,
+        resourceId=delete_resource_id,
+        httpMethod="GET",
+        integrationHttpMethod="DELETE",
+        type="AWS",
+        uri=f"arn:aws:apigateway:{region_name}:s3:path/{bucket}/{object_name}",
+        credentials=role_arn,
+    )
+
+    aws_client.apigateway.put_integration_response(
+        restApiId=api_id, resourceId=get_resource_id, httpMethod="GET", statusCode="200"
+    )
+    aws_client.apigateway.put_integration_response(
+        restApiId=api_id, resourceId=put_resource_id, httpMethod="GET", statusCode="200"
+    )
+    aws_client.apigateway.put_integration_response(
+        restApiId=api_id, resourceId=delete_resource_id, httpMethod="GET", statusCode="200"
+    )
+
+    aws_client.apigateway.create_deployment(restApiId=api_id, stageName=stage_name)
+
+    get_invoke_url = api_invoke_url(api_id, stage_name, path="/get")
+    put_invoke_url = api_invoke_url(api_id, stage_name, path="/put")
+    delete_invoke_url = api_invoke_url(api_id, stage_name, path="/delete")
+
+    def _invoke(url, json_response: bool = False):
+        response = requests.get(url=url)
+        assert response.status_code == 200
+        if json_response:
+            return response.json()
+        return response
+
+    retry(lambda: _invoke(put_invoke_url), retries=10, sleep=2)
+    get_object = retry(lambda: _invoke(get_invoke_url, json_response=True), retries=10, sleep=3)
+    snapshot.match("get-object", get_object)
+    _invoke(delete_invoke_url)
+
+    get_object = retry(lambda: _invoke(get_invoke_url), retries=10, sleep=2)
+    snapshot.match("get-deleted-object", xmltodict.parse(get_object.content))
```

Comment:
it seems this snapshot hasn't been regenerated, as it's the raw XML in the snapshot. We might need some transformers on `HostId` and `RequestId` I think, but should be good.

Reply:
It has been re-snapped and also removed some previous snapshots now useless.